src/category_theory/preadditive/yoneda.lean

```diff
@@ -0,0 +1,97 @@
+
+## TODO
+* The Yoneda embedding is additive itself
+* The Yoneda embedding for preadditive categories relates to the Yoneda embedding for linear
+  categories.
+
+-/
+
+universes v u
+
+open category_theory.preadditive opposite
+
+namespace category_theory
+
+variables {C : Type u} [category.{v} C] [preadditive C]
+
+/--
+The Yoneda embedding for preadditive categories sends an object `Y` to the presheaf sending an
+object `X` to `End Y`-module of morphisms `X ⟶ Y`.
+-/
+@[simps]
+def preadditive_yoneda_obj (Y : C) : Cᵒᵖ ⥤ Module.{v} (End Y) :=
+{ obj := λ X, Module.of _ (X.unop ⟶ Y),
+  map := λ X X' f,
+  { to_fun := λ g, f.unop ≫ g,
+    map_add' := λ g g', comp_add _ _ _ _ _ _,
+    map_smul' := λ r g, eq.symm $ category.assoc _ _ _ } }
+
+/--
+The Yoneda embedding for preadditive categories sends an object `Y` to the presheaf sending an
+object `X` to the group of morphisms `X ⟶ Y`. At each point, we get an additional `End Y`-module
+structure, see `preadditive_yoneda_obj`.
+-/
+@[simps]
+def preadditive_yoneda : C ⥤ (Cᵒᵖ ⥤ AddCommGroup.{v}) :=
+{ obj := λ Y, preadditive_yoneda_obj Y ⋙ forget₂ _ _,
+  map := λ Y Y' f,
+  { app := λ X,
+    { to_fun := λ g, g ≫ f,
+      map_zero' := limits.zero_comp,
+      map_add' := λ g g', add_comp _ _ _ _ _ _ },
+    naturality' := λ X X' g, AddCommGroup.ext _ _ _ _ $ λ x, category.assoc _ _ _ },
+  map_id' := λ X, by { ext, simp },
+  map_comp' := λ X Y Z f g, by { ext, simp } }
+
+/--
+The Yoneda embedding for preadditive categories sends an object `X` to the copresheaf sending an
+object `Y` to `End X`-module of morphisms `X ⟶ Y`.
+-/
+@[simps]
+def preadditive_coyoneda_obj (X : Cᵒᵖ) : C ⥤ Module.{v} (End X) :=
+{ obj := λ Y, Module.of _ (unop X ⟶ Y),
+  map := λ Y Y' f,
+  { to_fun := λ g, g ≫ f,
+    map_add' := λ g g', add_comp _ _ _ _ _ _,
+    map_smul' := λ r g, category.assoc _ _ _ } }
+
+/--
+The Yoneda embedding for preadditive categories sends an object `X` to the copresheaf sending an
+object `Y` to the group of morphisms `X ⟶ Y`. At each point, we get an additional `End X`-module
+structure, see `preadditive_coyoneda_obj`.
+-/
+@[simps]
+def preadditive_coyoneda : Cᵒᵖ ⥤ (C ⥤ AddCommGroup.{v}) :=
+{ obj := λ X, preadditive_coyoneda_obj X ⋙ forget₂ _ _,
+  map := λ X X' f,
+  { app := λ Y,
+    { to_fun := λ g, f.unop ≫ g,
+      map_zero' := limits.comp_zero,
+      map_add' := λ g g', comp_add _ _ _ _ _ _ },
+    naturality' := λ Y Y' g, AddCommGroup.ext _ _ _ _ $ λ x, eq.symm $ category.assoc _ _ _ },
+  map_id' := λ X, by { ext, simp },
+  map_comp' := λ X Y Z f g, by { ext, simp } }
+
+instance additive_yoneda_obj (X : C) : functor.additive (preadditive_yoneda_obj X) := {}
+instance additive_yoneda_obj' (X : C) : functor.additive (preadditive_yoneda.obj X) := {}
+instance additive_coyoneda_obj (X : Cᵒᵖ) : functor.additive (preadditive_coyoneda_obj X) := {}
+instance additive_coyoneda_obj' (X : Cᵒᵖ) : functor.additive (preadditive_coyoneda.obj X) := {}
+
```

Comment:
Please include compatibility with the "normal" Yoneda embedding, after composing with the forgetful functor `AddCommGroup ⥤ Type`.

Reply:
Done.